raster/src/main/scala/geotrellis/raster/reproject/RasterRegionReproject.scala

```diff
@@ -109,74 +134,140 @@ object RasterRegionReproject {
+          val (pxs, xs, ys) = rowcoords(i)
+          cfor(0)(_ < xs.size, _ + 1) { s =>
+            target.setDouble(pxs(s), i, resampler.resampleDouble(xs(s), ys(s)))
           }
-        } else {
-          cfor(0)(_ < xs.size, _ + 1){ s =>
-            buffer.set(pxs(s), i, resampler.resample(xs(s), ys(s)))
+        }
+      } else {
+        cfor(0)(_ < rasterExtent.rows, _ + 1){ i =>
+          val (pxs, xs, ys) = rowcoords(i)
+          cfor(0)(_ < xs.size, _ + 1) { s =>
+            target.set(pxs(s), i, resampler.resample(xs(s), ys(s)))
           }
         }
       }
     }
   }
 
   implicit val multibandInstance = new RasterRegionReproject[MultibandTile] {
-    def regionReproject(raster: Raster[MultibandTile], src: CRS, dest: CRS, rasterExtent: RasterExtent, region: Polygon, resampleMethod: ResampleMethod): Raster[MultibandTile] = {
+    def regionReproject(
+      raster: Raster[MultibandTile], 
+      src: CRS, 
+      dest: CRS, 
+      rasterExtent: RasterExtent, 
+      region: Polygon, 
+      resampleMethod: ResampleMethod
+    ): Raster[MultibandTile] = {
       val bands = Array.ofDim[MutableArrayTile](raster.tile.bandCount)
       cfor(0)(_ < bands.length, _ + 1) { i =>
-        bands(i) = raster.tile.band(i).prototype(rasterExtent.cols, rasterExtent.rows).mutable
+        bands(i) = ArrayTile.empty(raster.tile.band(i).cellType, rasterExtent.cols, rasterExtent.rows)
       }
-      mutableRegionReproject(MultibandTile(bands), raster, src, dest, rasterExtent, region, resampleMethod)
+      reprojectToBuffer(raster, src, dest, bands, rasterExtent, region, resampleMethod)
       Raster(MultibandTile(bands), rasterExtent.extent)
     }
 
-    def mutableRegionReproject(target: MultibandTile, raster: Raster[MultibandTile], src: CRS, dest: CRS, rasterExtent: RasterExtent, region: Polygon, resampleMethod: ResampleMethod) = {
-      val trans = Proj4Transform(dest, src)
+    def regionReprojectMutable(
+      raster: Raster[MultibandTile], 
+      src: CRS, 
+      dest: CRS, 
+      target: Raster[MultibandTile], 
+      region: Polygon, 
+      resampleMethod: ResampleMethod
+    ): Raster[MultibandTile] = {
+      val bufferRE = target.rasterExtent
       val bands = Array.ofDim[MutableArrayTile](raster.tile.bandCount)
-
       cfor(0)(_ < bands.length, _ + 1) { i =>
-        bands(i) = target.band(i).mutable
+        bands(i) = target.tile.band(i).mutable
       }
 
-      val resampler = (0 until raster.tile.bandCount).map { i =>
-        Resample(resampleMethod, raster.tile.band(i), raster.extent, raster.rasterExtent.cellSize)
+      reprojectToBuffer(raster, src, dest, bands, bufferRE, region, resampleMethod)
+      Raster(MultibandTile(bands), bufferRE.extent)
+    }
+
+    private def reprojectToBuffer(raster: Raster[MultibandTile], src: CRS, dest: CRS, buffer: Array[MutableArrayTile], rasterExtent: RasterExtent, region: Polygon, resampleMethod: ResampleMethod): Unit = {
```

Comment:
These could be DRYied out with single-band case, but might as well include the most specific version with the least conditionals